### PR TITLE
PLATUI-1123 Add timeline component

### PR DESCRIPTION
**Pull request by natcarey**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Aliases.scala [hunk 1]
```diff
@@ -101,6 +101,9 @@ trait Aliases {
 
   type AddToAList = viewmodels.addtoalist.AddToAList
   val AddToAList = viewmodels.addtoalist.AddToAList
+
+  type Timeline = viewmodels.timeline.Timeline
```
Comment (matthewmascord) [inline src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Aliases.scala:105]: Do we need an alias for Event too?

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcTimeline.scala.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.hmrcfrontend.views.viewmodels.timeline.Timeline
+
+@this()
+
+@(params: Timeline)(implicit messages: Messages)
+@import params._
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcTimeline.scala.html:22]: Probably worth enforcing mandatory parameters like here: https://github.com/hmrc/play-frontend-hmrc/blob/7bc0366ad080c1c24be2d6aa328721e186e8f61d/src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcCharacterCount.scala.html#L29

Comment (natcarey) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/hmrcTimeline.scala.html:22]: I'm not sure what we'd test for there, `headingLevel` has a default, `events` defaults to an empty sequence.  The `Event` requires a `title`, `time` and `content` with an optional `datetime`.

File: src/main/scala/uk/gov/hmrc/hmrcfrontend/views/viewmodels/timeline/Event.scala [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.views.viewmodels.timeline
+
+import play.api.libs.json.{Json, OFormat}
+
+case class Event(
+  title: String,
```
[the remark this answers is not in the document]
Comment (natcarey) [reply src/main/scala/uk/gov/hmrc/hmrcfrontend/views/viewmodels/timeline/Event.scala:22]: I don't think there is a logical default value that we'd want in production - if we want a value to test with I'd suggest we add one when we test rather that writing one which will hit production.

[the remark this answers is not in the document]
Comment (natcarey) [reply src/main/scala/uk/gov/hmrc/hmrcfrontend/views/viewmodels/timeline/Event.scala:22]: I don't see defaults in the `addToAList` e.g.

```
case class ListItem(
  name: String,
  changeUrl: String,
  removeUrl: String
)
```

File: src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Aliases.scala (hunk 1 above)
Comment (natcarey) [reply src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Aliases.scala:105]: Updated here https://github.com/hmrc/play-frontend-hmrc/pull/77/commits/ea82f4b7c81cccaede4d1d55f2701f17e7893337

**Review by matthewmascord — APPROVED** [review]

**Review by oscarduignan — APPROVED** [review]
looks good to me!

**Pull request merged**